tests/integration/memgraph.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2016-2020 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "mgclient.h"
+extern "C" {
+#include "mgvalue.h"
+}
+
+template <typename T>
+T GetEnvOrDefault(const std::string &value_name, const T &default_value) {
+  const char *char_value = std::getenv(value_name.c_str());
+  if (char_value) {
+    T value;
+    std::stringstream env_value_stream(char_value);
+    env_value_stream >> value;
+    return value;
+  }
+  return default_value;
+}
+
+std::string GetStringValue(const mg_string *value) {
+  return std::string(value->data, value->size);
+}
+
+std::string GetStringValue(const mg_value *value) {
+  const mg_string *s = mg_value_string(value);
+  return GetStringValue(s);
+}
```

Comment:
To simplify the comments below, add functions that convert a `const mg_value *` to an integer, node, relationship, map... and make them perform the mandatory type checking so that you don't have to explicitly check each type in the logic below.